app/controllers/short_url_requests_controller.rb

```diff
@@ -49,6 +51,21 @@ def reject
     redirect_to short_url_requests_path
   end
 
+  def remove
+    router.delete_route("#{@short_url_request.from_path}", "exact")
```

Comment:
The router-api has [just been simplified](https://github.com/alphagov/router-api/pull/61) so that it's no longer necessary to specify the route type when deleting (or querying) a route.

There's an as yet unreleased version of gds-api-adapters that includes this change.  PR coming shortly to release this.